src/data/docs/language-reference/README.md

```diff
@@ -911,10 +911,10 @@ The type `{Store v}` means that the computation which results in that type requi
 A constructor `{A} T` for some ability `A` and some type `T`  (or a function which uses such a constructor), can only be used in a scope where the ability `A` is provided. Abilities are provided by `handle` expressions:
 
 ``` unison
-handle h in x
+handle e with h
 ```
 
-This expression gives `x` access to abilities handled by the function `h` which must have the type `Request A T -> T` if `x` has type `{A} T`. The type constructor `Request` is a special builtin provided by Unison which will pass arguments of type `Request A T` to a handler for the ability `A`.
+This expression gives `e` access to abilities handled by the function `h`. If `h` has type `Request A T1 -> T2` and the expression `e` has type `{A} T1`, then `handle e with h` will have type `T2`. The type constructor `Request` is a special builtin provided by Unison which will pass arguments of type `Request A T` to a handler for the ability `A`. Note that `h` _must_ accept an argument of type `Request A T` if it is used to handle `e` of type `{A} T1`.
```

Comment:
@aryairani Can you check that this new version is accurate? It's more verbose than before, but I was trying to make it clear _and_ precise. There's probably still room for improvement.